### ISSUE TITLE
Deal with vite application base path

### ISSUE THREAD

**Issue by smoothdub** [quarkiverse/quarkus-quinoa#299]
Hello,

My quarkus application is running on a specific root path. Let's say /hello.
That's the same for my vite react application which has the base path set at /hello.

It looks like I cannot configure Quinoa to set this base endpoint for my application.
It removes the root path and calls the front without it anyway.

Did I miss something or I should do a PR for that?
(Working on Quarkus 2 btw)
Thanks in advance for your reply.

**Comment by melloware** [comment]
So I am confused would your UI then run at `/hello/hello` ?

**Comment by melloware** [comment]
Can you provide a small standalong app showing the issue or paste your `vite.config` here so I can try it?

**Comment by smoothdub** [comment]
Here is a playground: https://github.com/smoothdub/quinoa-vite-test
You should everything in the readme :)

**Comment by melloware** [comment]
Thanks let me check it out!

**Comment by ia3andy** [comment]
Could you give a bit of context of the use case for this to be needed.

In most scenario, the api is on it's own path (i.e. `/api`), then the frontend is accessible on any other path starting from the root.

I am interested to understand why you would want to do the opposite :)

It would be nice to have some example production urls (with fake domains) that represent your use case.

**Comment by smoothdub** [comment]
I'm using Quarkus to serve both frontend and backend. As my application must run behind a reverse proxy, I need to set a root path to my quarkus application.  Behind this root path. I have indeed some specific paths like /API which is reserved for API calls.
Other calls muster be rerouted to / and will be treated by the spa. 
When building this Spa, I need to give it a base path too. Otherwise, CSS and JS resolution won't work. 

Please check the read me of the project posted above to see what are my expectations. 
Thanks!

**Comment by ia3andy** [comment]
> I'm using Quarkus to serve both frontend and backend. As my application must run behind a reverse proxy, I need to set a root path to my quarkus application. Behind this root path. I have indeed some specific paths like /API which is reserved for API calls. Other calls muster be rerouted to / and will be treated by the spa. When building this Spa, I need to give it a base path too. Otherwise, CSS and JS resolution won't work.
> 
> Please check the read me of the project posted above to see what are my expectations. Thanks!

Well, that's the standard way of doing thing. You should have to put the UI on a different subpath, it's just a question of route order which is already handled by Quinoa.

If you are using routing in front of it, it's again just a question of route order.

**Comment by smoothdub** [comment]
> > I'm using Quarkus to serve both frontend and backend. As my application must run behind a reverse proxy, I need to set a root path to my quarkus application. Behind this root path. I have indeed some specific paths like /API which is reserved for API calls. Other calls muster be rerouted to / and will be treated by the spa. When building this Spa, I need to give it a base path too. Otherwise, CSS and JS resolution won't work.
> > Please check the read me of the project posted above to see what are my expectations. Thanks!
>  
> Well, that's the standard way of doing thing. You should have to put the UI on a different subpath, it's just a question of route order which is already handled by Quinoa.
> 
> If you are using routing in front of it, it's again just a question of route order.

Not sure that you got the point. If I put the front on another subpath, it will still try to get my ui on / ( dev server side) .
I haven't see any property to change that behavior.

**Comment by melloware** [comment]
I think I see what you are saying.  You could propose a PR as I could not get this working with your current configuration.

**Comment by smoothdub** [comment]
> I think I see what you are saying. You could propose a PR as I could not get this working with your current configuration.

Thanks for the double check !
I'll check tomorrow to make it work

**Comment by ia3andy** [comment]
@smoothdub I still don't get why you would need a different path for the UI. Try the default behavior with the api behind `/api`, everything is served as it should be without any problem (js, ...) even SPA routing works. Why would it be any different behind a reverse proxy (nginx?)?

**Comment by ia3andy** [comment]
Anyway :), I would be ok having a new config `quarkus.quinoa.path` to have the ui on a specific subpath. This means the SPA needs to be setup with the sub-path (ex: PUBLIC_URL with RCA).

AFAIK, in your situation, this will be a problem because your app in production will not use the sub-path as you are rewriting to `/`.

**Comment by smoothdub** [comment]
@ia3andy The default behavior works well indeed but do not suit me for this specific use case.
I **must** serve my whole application behind a specific root path. We have one reverse proxy for several apps. 

quarkus root path: /app 
API path: /app/api

Then my web app served by quinoa must be exposed behind /app too.
That's indeed the default behavior.
If I try to reach the web app on /app , it retrieves correctly the html.
But then I need to fetch the CSS and the JS. If I don't put a base path to react, It will fetch the CSS and the JS on / and not /app.

As far as I know, nginx cannot rewrite the <link> I'm my html.

**Comment by ia3andy** [comment]
> @ia3andy The default behavior works well indeed but do not suit me for this specific use case. I **must** serve my whole application behind a specific root path. We have one reverse proxy for several apps.
> 
> quarkus root path: /app API path: /app/api
> 
> Then my web app served by quinoa must be exposed behind /app too. That's indeed the default behavior. If I try to reach the web app on /app , it retrieves correctly the html. But then I need to fetch the CSS and the JS. If I don't put a base path to react, It will fetch the CSS and the JS on / and not /app.
> 
> As far as I know, nginx cannot rewrite the I'm my html.

Ok so what you are looking for is this I believe: https://stackoverflow.com/questions/49429906/how-should-i-configure-create-react-app-to-serve-app-from-subdirectory

**Comment by melloware** [comment]
I agree with @ia3andy I think you just need to tweak your Vite App and not Quinoa.

**Comment by gbourant** [comment]
I have created a web application that utilizes the quarkus-quinoa extension for an e-shop and an Angular application for the admin dashboard. However, I am facing an issue where I cannot serve both the quinoa and Angular app through the quarkus app without using a reverse proxy. This is because the quarkus-quinoa app is currently served from the root path "/" by default.

To address this issue, I request to add support for configuring the path from which the quarkus-quinoa app is served. This will allow users to serve both the quinoa and Angular apps through the same quarkus app without any conflicts.

Thank you.

```java
@Path("/")
@ApplicationScoped
public class HtmlResource {

    @Inject
    Template index;

    @Inject
    Template cart;

    @GET
    public TemplateInstance getIndex() {
        return index.instance();
    }

    @GET
    @Path("cart")
    public TemplateInstance getContant() {
        return cart.instance();
    }

}
```

**Comment by ia3andy** [comment]
> I have created a web application that utilizes the quarkus-quinoa extension for an e-shop and an Angular application for the admin dashboard. However, I am facing an issue where I cannot serve both the quinoa and Angular app through the quarkus app without using a reverse proxy. This is because the quarkus-quinoa app is currently served from the root path "/" by default.
> 
> To address this issue, I request to add support for configuring the path from which the quarkus-quinoa app is served. This will allow users to serve both the quinoa and Angular apps through the same quarkus app without any conflicts.
> 
> Thank you.
> 
> ```java
> @Path("/")
> @ApplicationScoped
> public class HtmlResource {
> 
>     @Inject
>     Template index;
> 
>     @Inject
>     Template cart;
> 
>     @GET
>     public TemplateInstance getIndex() {
>         return index.instance();
>     }
> 
>     @GET
>     @Path("cart")
>     public TemplateInstance getContant() {
>         return cart.instance();
>     }
> 
> }
> ```

Could you create a new issue "Allow to serve Quinoa app on a specific path", this one is a different problem.
Also could you give a bit more detail, since Angular can also be served by Quinoa, it makes the description hard to understand.

**Comment by gbourant** [comment]
> > I have created a web application that utilizes the quarkus-quinoa extension for an e-shop and an Angular application for the admin dashboard. However, I am facing an issue where I cannot serve both the quinoa and Angular app through the quarkus app without using a reverse proxy. This is because the quarkus-quinoa app is currently served from the root path "/" by default.
> > To address this issue, I request to add support for configuring the path from which the quarkus-quinoa app is served. This will allow users to serve both the quinoa and Angular apps through the same quarkus app without any conflicts.
> > Thank you.
> > ```java
> > @Path("/")
> > @ApplicationScoped
> > public class HtmlResource {
> > 
> >     @Inject
> >     Template index;
> > 
> >     @Inject
> >     Template cart;
> > 
> >     @GET
> >     public TemplateInstance getIndex() {
> >         return index.instance();
> >     }
> > 
> >     @GET
> >     @Path("cart")
> >     public TemplateInstance getContant() {
> >         return cart.instance();
> >     }
> > 
> > }
> > ```
> 
> @gbourant Could you create a new issue "Allow to serve Quinoa app on a specific path", this one is a different problem. Also could you give a bit more detail, since Angular can also be served by Quinoa, it makes the description hard to understand.

@ia3andy The issue is not related to any specific JavaScript framework such as Angular, React, or Vue. I believe I have provided sufficient explanation for this in the new issue I raised, #302.

**Comment by melloware** [comment]
1.2.9 has been released with this fix.
